### ISSUE TITLE
usage of polymer

### ISSUE THREAD

**Issue by ChristophCaina** [frozenwizard/onlylocklock#16]
Hi, Since HA 2023.04 I am getting the following information in my HA Logs:

```
Logger: frontend.js.latest.202304111
Source: components/system_log/__init__.py:257
First occurred: 22. April 2023 um 08:34:12 (4 occurrences)
Last logged: 22. April 2023 um 23:54:50

WARNING: Polymer will be removed from window in Home Assistant 2023.5. More info: https://developers.home-assistant.io/blog/2023/04/04/deprecating_polymer
```

After some research, I found that onlyLockLock seems to use Polymer in some functions:

```
 W:\www\community\onlylocklock\only-arm-alarm-row.js (2 Treffer)
	Row  4: export class OnlyArmAlarmRow extends Polymer.Element {
	Row  6:     return Polymer.html`
  W:\www\community\onlylocklock\only-close-cover-row.js (2 Treffer)
	Row  4: export class OnlyCloseCoverRow extends Polymer.Element {
	Row  6:     return Polymer.html`
  W:\www\community\onlylocklock\only-lock-lock-row.js (2 Treffer)
	Row  4: export class OnlyLockLockRow extends Polymer.Element {
	Row  6:     return Polymer.html`
```

can you confirm if this might be the reason for getting the warning in my logs?
And would you consider to change this in order to be compatible with the upcomming 2023.05 version of HA?

Thanks

**Comment by frozenwizard** [comment]
Hey I will take a look at that in a day or so.

**Comment by WhimsySpoon** [comment]
Can confirm that as of HA Core 2023.5.x, this card no longer loads as Polymer support has been removed.

**Comment by frozenwizard** [comment]
Sorry for the wait, personal things came up.

I've created the 0.6 release which contains the removal of polymer.  So far it works on my end, please test and let me know everything works.

**Comment by WhimsySpoon** [comment]
Thanks for getting on this, and there's no need go apologise; personal life comes first!

I've just installed 0.6 and it's still refusing to load. The log is showing:

`https://xxxxxx.ui.nabu.casa/hacsfiles/onlylocklock/only-arm-alarm-row.js:4:38 Uncaught ReferenceError: Polymer is not defined`

**Comment by frozenwizard** [comment]
That's weird.  I checked the code, then downloaded the release.  Line 4 does not have Polymer in it.  Grepping the codebase I left in a Polymer reference for only close cover.   Which I've got a fix for now in main.  I haven't upgraded to the May Home Assistant release as some of the other repos didn't remove polymer as well so I didn't catch this.  I'm installing that now to make sure.

My suspicion is something got cached in your browser.  Usually if you enable developer mode in your browser it has an option to not cache things.  If you're using the iOS app, there's a setting in there as well to force reload it (or something to that effect).

If that doesn't work, try selecting the main branch as it will include that above fix.

**Comment by frozenwizard** [comment]
I upgraded HA, the page was broken with the un updated only lock.  Upgraded, it was still broken, went into developer mode and ignored the cache and the page loaded correctly.

**Comment by frozenwizard** [comment]
You can use the v0.6.1 release.

**Comment by WhimsySpoon** [comment]
Thanks. I did try disabling the cache and Ctrl+F5, even in different devices and it didn't work. 0.6.1 doesn't work either. I suspect its something with my configuration rather than your code. I'll have a dig around in my setup and see if I can find out what's happening.

**Comment by WhimsySpoon** [comment]
The error seems to be complaing about the alarm rather than lock.

I'm using custom:only-lock-lock-row, but the error is `https://xxxxx.ui.nabu.casa/hacsfiles/onlylocklock/only-arm-alarm-row.js:4:38 Uncaught ReferenceError: Polymer is not defined`

**Comment by WhimsySpoon** [comment]
Ok, I've got it working. I had to manually clear the cache on three files (in the order below):

- only-arm-alarm-row.js
- only-close-cover-row.js
- only-lock-lock-row.js

This should have been taken care of when I disabled cache and/or Ctrl+F5, but didn't. I even tried a restart of HA Core just for good measure. I'll put this down to Chrome.

I just need to work out how to do this in the Android Companion app now.
